vortex-dtype/src/arrow.rs

```diff
@@ -1,25 +1,152 @@
-//! Convert between Vortex [`crate::DType`] and Apache Arrow [`arrow_schema::DataType`].
+//! Convert between Vortex [`DType`] and Apache Arrow [`DataType`].
 //!
-//! Apache Arrow's type system includes physical information, which could lead to ambiguities as
-//! Vortex treats encodings as separate from logical types.
+//! ## Arrow -> Vortex
 //!
-//! [`DType::to_arrow_schema`] and its sibling [`DType::to_arrow_dtype`] use a simple algorithm,
-//! where every logical type is encoded in its simplest corresponding Arrow type. This reflects the
-//! reality that most compute engines don't make use of the entire type range arrow-rs supports.
+//! Each Arrow `DataType` has a defined mapping onto its nearest Vortex data type, via
+//! implementation of the `FromArrowType` trait.
 //!
-//! For this reason, it's recommended to do as much computation as possible within Vortex, and then
-//! materialize an Arrow ArrayRef at the very end of the processing chain.
+//! All of the Arrow primitive types map onto the equivalent Vortex primitives:
+//!
+//! ```rust
```

Comment:
@gatesn curious for your thoughts on this design, particularly the `ArrowTypeConversion` plugin trait.

I realized that I was wrong about needing to change `DType::to_arrow`'s signature, because the only place Arrow is able to reason about extension types anyway is in the context of a full struct `Schema`. So instead I ended up just making this plugin trait that is used to convert ExtDType children of Vortex Struct types.

I'm also wondering if this should be broadened so that `ToArrowTemporal` could actually just be replaced by this as well, in which case we'd return not a metadata hashmap but a full arrow `Field` type.